src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -151,3 +157,21 @@ class BlockingProcess(
     else
       args
 }
+
+object StreamGuardian {
+  def props(closeable: Closeable): Props =
+    Props(new StreamGuardian(closeable))
+}
+
+/**
+ * Responsible for closing a stream in order to free up any blocked
+ * threads that attempt to read from the stream.
+ */
+class StreamGuardian(closeable: Closeable) extends Actor {
```

Comment:
Should not this watch the approrpriate other actor so that if it stops then this will also stop?